src/test/java/com/redhat/devtools/lsp4ij/features/semanticTokens/viewProvider/TypeScriptSemanticTokensFileViewProviderTest.java

```diff
@@ -0,0 +1,190 @@
+                8,
+                9,
+                1,
+                3,
+                11,
+                3,
+                11,
+                3,
+                1,
+                8,
+                7,
+                7,
+                16,
+                0,
+                8,
+                3,
+                11,
+                16,
+                1,
+                14,
+                11,
+                7,
+                41,
+                0,
+                14,
+                4,
+                7,
+                16,
+                0,
+                5,
+                2,
+                9,
+                24,
+                1,
+                8,
+                7,
+                7,
+                16,
+                0,
+                8,
+                3,
+                11,
+                16,
+                0,
+                4,
+                11,
+                7,
+                40
+              ]
+            }
+            """;
+
+    public TypeScriptSemanticTokensFileViewProviderTest() {
+        super("*.ts");
+    }
+
+    public void testEnabled() {
+        assertViewProviderEnabled(
+                TEST_FILE_NAME,
+                TEST_FILE_BODY,
+                MOCK_SEMANTIC_TOKENS_PROVIDER_JSON,
+                MOCK_SEMANTIC_TOKENS_JSON,
+                Map.ofEntries(
+                        Map.entry(fileBody -> fileBody.indexOf("Foo"), LSPSemanticTokenElementType.DECLARATION),
+                        Map.entry(fileBody -> fileBody.indexOf("field"), LSPSemanticTokenElementType.DECLARATION),
+                        Map.entry(fileBody -> fileBody.indexOf("property"), LSPSemanticTokenElementType.DECLARATION),
+                        Map.entry(fileBody -> fileBody.indexOf("bar()"), LSPSemanticTokenElementType.DECLARATION),
+                        Map.entry(fileBody -> fileBody.indexOf("console.log('"), LSPSemanticTokenElementType.REFERENCE),
+                        Map.entry(fileBody -> fileBody.indexOf("log('"), LSPSemanticTokenElementType.REFERENCE),
+                        Map.entry(fileBody -> fileBody.indexOf("declaration"), LSPSemanticTokenElementType.DECLARATION),
+                        Map.entry(fileBody -> fileBody.indexOf("Math"), LSPSemanticTokenElementType.REFERENCE),
+                        Map.entry(fileBody -> fileBody.indexOf("PI"), LSPSemanticTokenElementType.REFERENCE),
+                        Map.entry(fileBody -> fileBody.indexOf("console.log(d"), LSPSemanticTokenElementType.REFERENCE),
+                        Map.entry(fileBody -> fileBody.indexOf("log(d"), LSPSemanticTokenElementType.REFERENCE),
+                        Map.entry(fileBody -> fileBody.indexOf("declaration)"), LSPSemanticTokenElementType.REFERENCE)
+                )
+        );
+    }
+
+    // Confirms the behavior for a language server that doesn't support semantic tokens or when they're not yet present
+    public void testEnabledNoSemanticTokens() {
```

Comment:
Confirms behavior when the feature is enabled against a language server that doesn't support semantic tokens, or when no semantic tokens have yet been added to the view provider.

The CSS language server does not support semantic tokens, but by default CSS is an `AbstractFileType` and not TextMate, so setting up that use case is more complex than is warranted when we can just pass null mock semantic tokens responses.